integration-tests/src/test/java/com/datastax/oss/driver/api/core/metadata/TokenITBase.java

```diff
@@ -339,4 +343,16 @@ public void should_create_token_from_partition_key() {
     assertThat(tokenMap.newToken(TypeCodecs.INT.encodePrimitive(1, protocolVersion)))
         .isEqualTo(expected);
   }
+
+  private TokenMap getTokenMap() {
+    Optional<? extends TokenMap> maybeTokenMap = session().getMetadata().getTokenMap();
+    if (maybeTokenMap.isPresent()) {
```

Comment:
Could be simplified to:

```
assertThat(maybeTokenMap).isPresent().hasValueSatisfying(tokenMap -> {
  assertThat(tokenMap.getPartitionerName()).isEqualTo(expectedPartitionerName);
});
```

Reply:
"simplified"? 😉 
More seriously, I had also considered:
```java
    return session()
        .getMetadata()
        .getTokenMap()
        .map(
            tokenMap -> {
              assertThat(tokenMap.getPartitionerName()).isEqualTo(expectedPartitionerName);
              return tokenMap;
            })
        .orElseThrow(() -> new AssertionError("Expected token map to be present"));
```
which has a bit more of a "functional" flavor but I still find it about as ugly as the imperative version.
I'm often disappointed by `Optional` code, without algebraic types or pattern matching you can't fully take advantage of it.

Reply:
Well, 4 lines instead of 8 is a simplification to me ;-) but it doesn't matter really, the current version is fine enough.